### ISSUE TITLE
Unable to start vmware in trial mode

### ISSUE THREAD

**Issue by ericrhenry5488** [stefantalpalaru/gentoo-overlay#88]
After emerging the latest vmware-workstation ebuild and running the --config as instructed, starting up vmware as a user assigned to the vmware group brings up the graphical configurator. At startup, the terminal output is:

[AppLoader] Use shipped PC/SC Lite smart card framework.
An up-to-date "pcsc-lite-libs" or "libpcsclite1" package from your system is preferred.
I/O warning : failed to load external entity "/etc/vmware/hostd/proxy.xml"

When I select the trial option, and click 'Finish', terminal output is:

Error getting authority: Error initializing authority: Could not connect: No such file or directory

after which nothing happens. Repeated clicks just generate a repeat of the above message. I am able to exit from the configurator using Cancel, so things have not frozen.

I'm wondering if there is something still missing, or mis-configured, from my install, or perhaps there is a path problem somewhere. I have done some preliminary probes using 'strace', with no enlightening results so far, but I will keep trying.

I don't want to try to virgin install using the downloaded *.bundle file, if I can avoid it, out of huge respect and appreciation for the effort put in to create this ebuild--and because the process of wrangling the various required components looks potentially painful.

Many thanks,
Eric

Output of

emerge --info vmware-workstation follows (excuse the length):

Portage 3.0.17 (python 3.8.8-final-0, default/linux/amd64/17.1/desktop/plasma, gcc-10.2.0, glibc-2.32-r7, 5.10.27-gentoo x86_64)
=================================================================
                         System Settings
=================================================================
System uname: Linux-5.10.27-gentoo-x86_64-Intel-R-_Core-TM-_i7-10510U_CPU_@_1.80GHz-with-glibc2.2.5
KiB Mem:    16046952 total,  12591740 free
KiB Swap:   32766972 total,  32766972 free
Timestamp of repository gentoo: Sat, 17 Apr 2021 18:30:01 +0000
Head commit of repository gentoo: e0ed6349675a90cbe88b18e97c6f62aeda679759
sh bash 5.0_p18
ld GNU ld (Gentoo 2.35.2 p1) 2.35.2
app-shells/bash:          5.0_p18::gentoo
dev-lang/perl:            5.30.3::gentoo
dev-lang/python:          3.8.8_p1::gentoo, 3.9.2_p1::gentoo
dev-lang/rust:            1.47.0-r2::gentoo
dev-util/cmake:           3.18.5::gentoo
sys-apps/baselayout:      2.7::gentoo
sys-apps/openrc:          0.42.1-r1::gentoo
sys-apps/sandbox:         2.20::gentoo
sys-devel/autoconf:       2.13-r1::gentoo, 2.69-r5::gentoo
sys-devel/automake:       1.16.2-r1::gentoo
sys-devel/binutils:       2.35.2::gentoo
sys-devel/gcc:            10.2.0-r5::gentoo
sys-devel/gcc-config:     2.4::gentoo
sys-devel/libtool:        2.4.6-r6::gentoo
sys-devel/make:           4.3::gentoo
sys-kernel/linux-headers: 5.10::gentoo (virtual/os-headers)
sys-libs/glibc:           2.32-r7::gentoo
Repositories:

gentoo
    location: /var/db/repos/gentoo
    sync-type: rsync
    sync-uri: rsync://rsync.gentoo.org/gentoo-portage
    priority: -1000
    sync-rsync-verify-jobs: 1
    sync-rsync-verify-metamanifest: yes
    sync-rsync-verify-max-age: 24
    sync-rsync-extra-opts: 

stefantalpalaru
    location: /var/lib/layman/stefantalpalaru
    sync-type: laymansync
    sync-uri: https://github.com/stefantalpalaru/gentoo-overlay.git
    masters: gentoo
    priority: 50

ACCEPT_KEYWORDS="amd64"
ACCEPT_LICENSE="@FREE"
CBUILD="x86_64-pc-linux-gnu"
CFLAGS="-march=native -O2 -pipe"
CHOST="x86_64-pc-linux-gnu"
CONFIG_PROTECT="/etc /usr/lib64/libreoffice/program/sofficerc /usr/share/config /usr/share/gnupg/qualified.txt"
CONFIG_PROTECT_MASK="/etc/ca-certificates.conf /etc/dconf /etc/env.d /etc/fonts/fonts.conf /etc/gconf /etc/gentoo-release /etc/revdep-rebuild /etc/sandbox.d /etc/terminfo /etc/vmware-installer"
CXXFLAGS="-march=native -O2 -pipe"
DISTDIR="/var/cache/distfiles"
ENV_UNSET="CARGO_HOME DBUS_SESSION_BUS_ADDRESS DISPLAY GOBIN GOPATH PERL5LIB PERL5OPT PERLPREFIX PERL_CORE PERL_MB_OPT PERL_MM_OPT XAUTHORITY XDG_CACHE_HOME XDG_CONFIG_HOME XDG_DATA_HOME XDG_RUNTIME_DIR"
FCFLAGS="-march=native -O2 -pipe"
FEATURES="assume-digests binpkg-docompress binpkg-dostrip binpkg-logs config-protect-if-modified distlocks ebuild-locks fixlafiles ipc-sandbox merge-sync multilib-strict network-sandbox news parallel-fetch pid-sandbox preserve-libs protect-owned qa-unresolved-soname-deps sandbox sfperms strict unknown-features-warn unmerge-logs unmerge-orphans userfetch userpriv usersandbox usersync xattr"
FFLAGS="-march=native -O2 -pipe"
GENTOO_MIRRORS="https://gentoo.osuosl.org/ rsync://rsync.gtlib.gatech.edu/gentoo rsync://mirrors.rit.edu/gentoo/ http://gentoo.cs.utah.edu/"
LANG="C.UTF8"
LDFLAGS="-Wl,-O1 -Wl,--as-needed"
MAKEOPTS="-j4"
PKGDIR="/var/cache/binpkgs"
PORTAGE_CONFIGROOT="/"
PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --omit-dir-times --compress --force --whole-file --delete --stats --human-readable --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages --exclude=/.git"
PORTAGE_TMPDIR="/var/tmp"
USE="X a52 aac acl acpi activities alsa amd64 berkdb bluetooth branding bzip2 cairo cdda cdr cli crypt cups dbus declarative dri dts dvd dvdr elogind emboss encode eselect-ldso exif flac fortran gdbm gif gpm gtk gui iconv icu ipv6 jpeg kde kipi kwallet lcms libglvnd libnotify libtirpc mad mng mp3 mp4 mpeg multilib ncurses nls nptl ogg opengl openmp pam pango pcre pdf phonon plasma png policykit ppds qml qt5 readline sdl seccomp semantic-desktop spell split-usr ssl startup-notification svg tcpd tiff truetype udev udisks unicode upower usb vorbis widgets wxwidgets x264 xattr xcb xml xv xvid zlib" ABI_X86="64" ADA_TARGET="gnat_2018" ALSA_CARDS="ali5451 als4000 atiixp atiixp-modem bt87x ca0106 cmipci emu10k1x ens1370 ens1371 es1938 es1968 fm801 hda-intel intel8x0 intel8x0m maestro3 trident usb-audio via82xx via82xx-modem ymfpci" APACHE2_MODULES="authn_core authz_core socache_shmcb unixd actions alias auth_basic authn_alias authn_anon authn_dbm authn_default authn_file authz_dbm authz_default authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate dir disk_cache env expires ext_filter file_cache filter headers include info log_config logio mem_cache mime mime_magic negotiation rewrite setenvif speling status unique_id userdir usertrack vhost_alias" CALLIGRA_FEATURES="karbon sheets words" COLLECTD_PLUGINS="df interface irq load memory rrdtool swap syslog" CPU_FLAGS_X86="aes avx avx2 f16c fma3 mmx mmxext pclmul popcnt rdrand sse sse2 sse3 sse4_1 sse4_2 ssse3" ELIBC="glibc" GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock greis isync itrax mtk3301 nmea ntrip navcom oceanserver oldstyle oncore rtcm104v2 rtcm104v3 sirf skytraq superstar2 timing tsip tripmate tnt ublox ubx" INPUT_DEVICES="libinput" KERNEL="linux" LCD_DEVICES="bayrad cfontz cfontz633 glk hd44780 lb216 lcdm001 mtxorb ncurses text" LIBREOFFICE_EXTENSIONS="presenter-console presenter-minimizer" LUA_SINGLE_TARGET="lua5-1" LUA_TARGETS="lua5-1" OFFICE_IMPLEMENTATION="libreoffice" PHP_TARGETS="php7-3 php7-4" POSTGRES_TARGETS="postgres10 postgres11" PYTHON_SINGLE_TARGET="python3_8" PYTHON_TARGETS="python3_8" RUBY_TARGETS="ruby26" USERLAND="GNU" VIDEO_CARDS="intel i965 iris" XTABLES_ADDONS="quota2 psd pknock lscan length2 ipv4options ipset ipp2p iface geoip fuzzy condition tee tarpit sysrq proto steal rawnat logmark ipmark dhcpmac delude chaos account"
Unset:  CC, CPPFLAGS, CTARGET, CXX, EMERGE_DEFAULT_OPTS, INSTALL_MASK, LC_ALL, LINGUAS, PORTAGE_BINHOST, PORTAGE_BUNZIP2_COMMAND, PORTAGE_COMPRESS, PORTAGE_COMPRESS_FLAGS, PORTAGE_RSYNC_EXTRA_OPTS, RUSTFLAGS

=================================================================
                        Package Settings
=================================================================

app-emulation/vmware-workstation-16.1.1.17801498::stefantalpalaru was built with the following:
USE="cups doc modules vmware-tools-linux vmware-tools-windows -macos-guests -ovftool -systemd -vix -vmware-tools-darwin -vmware-tools-darwinPre15 -vmware-tools-linuxPreGlibc25 -vmware-tools-netware -vmware-tools-solaris -vmware-tools-winPre2k -vmware-tools-winPreVista" ABI_X86="(64)"

**Comment by stefantalpalaru** [comment]
It probably needs root access to write something to "/etc/vmware", but it usually prompts you for the root password using some bundled gksu library.

**Comment by ericrhenry5488** [comment]
Thanks, Stefan. Interestingly, when I start up under strace, the terminal output involving authority on each click of 'Finish' is replaced by

pkexec must be setuid root

(I don't really understand why the authority output doesn't appear when running under strace, and the word 'authority' appears nowhere in the voluminous strace output--except for probing .Xauthority.)

When I look at the strace output (following forks), each click appears to generate an instance of

execve("/usr/bin/pkexec", ["/usr/bin/pkexec", "/opt/vmware/lib/vmware/bin/vmware-setup-helper", "-n", "VMware Workstation", "-v", "16.0", "-s", "504RK-GV011-M88X2-0CAK6-2H2Q6"], 0x560b2bc22cb0 /* 68 vars */) = 0

The new PID runs through a few hundred system calls, without any clone() calls or the like, and exits with a status of 127. I Interpret that to mean that pkexec was unable to run the vmware-setup-helper for some reason. Regarding the 'setuid root' complaint, I see

$ ls -l /usr/bin/pkexec
-rws--x--x 1 root root 31264 Apr 17 19:03 /usr/bin/pkexec

and for the target program:

$ ls -l /opt/vmware/lib/vmware/bin/vmware-setup-helper
lrwxrwxrwx 1 root root 9 Apr 22 13:46 /opt/vmware/lib/vmware/bin/vmware-setup-helper -> appLoader

$ ls -l /opt/vmware/lib/vmware/bin/appLoader
-rwxr-xr-x 1 root root 880856 Apr 22 13:46 /opt/vmware/lib/vmware/bin/appLoader

Sorry if I'm missing something obvious...

Thanks again,
Eric

**Comment by ericrhenry5488** [comment]
Thanks, tried what you suggested. I get the same authority error. However, polkit-kde-agent was already emerged, but I re-merged it just in case. Same error. polkit has been emerged with the kde flag, re-merged that as well. No change in pkexec behavior. (FWIW, I get the same authority error when I run 'pkexec whoami' as root.)

I don't really know a lot about polkit; I do see that 'policykit' is one of my standard USE flags. Is there perhaps some other global flag I need? The 'No such file or directory' error message is a bit puzzling.

**Comment by ericrhenry5488** [comment]
A bit of a forehead-slapper. Another strace on pkaction revealed that dbus was not running. Have started that, now 'pkexec whoami' leads to an attempted authentication (but only on the terminal, no kde window). Unfortunately, entering the correct root password still leads to an authentication failure:

polkit-agent-helper-1: error response to PolicyKit daemon: GDBus.Error:org.freedesktop.PolicyKit1.Error.Failed: No session for cookie

Perhaps because polkit was merged with both gtk and kde?

**Comment by stefantalpalaru** [comment]
I have:

```text
[I] sys-auth/polkit
     Available versions:  0.116-r1^t (~)0.117^t 0.118^t {elogind examples gtk +introspection jit kde nls pam selinux systemd test}
     Installed versions:  0.118^t(05:25:08 PM 10/15/2020)(elogind gtk introspection nls pam -examples -jit -kde -selinux -systemd -test)
```

and it's actually `polkit-gnome` that handles the password window for me, not the KDE agent. I don't remember if I did something to pick one agent over another.

One interesting thing is that `strace pkexec whoami` makes it die without showing any window, so don't try to strace it.

**Comment by ericrhenry5488** [comment]
Curious about strace, never saw a program before that suffers from interacting with it.

Anyway, sorting out the dbus bit and rebooting--with the effect of restarting KDE plasma--seemed to do the trick. The kde-agent was called in, and I was able to run the installer. Apologies that the issue appears to have been a configuration issue with my machine, and I appreciate your patience.

During the installation of a Windows 10 guest, it flashed the message that vmware-tools could not be installed, and there is no evidence of vmware-tools anywhere in the guest. However, all options related to "installing/re-iinstalling vmware tools" seem to be grayed out in the settings menu. This is even after resetting the CD/DVD drive to point to the 'windows.iso' file that you provide. (As noted above, the emerge was done with vmware-tools-windows set.)

I've been googling around for (re)installing vmware-tools, but none of the suggestions have worked yet. It is apparently possible to download an in-guest installer for the tools, but it seems to be necessary to already have a customer account set up in order to do that.

I plan to close the primary issue I filed. (Again, my apologies.) But before I do that, do you have any insight about the tools problem?

Many thanks,
Eric

**Comment by stefantalpalaru** [comment]
> This is even after resetting the CD/DVD drive to point to the 'windows.iso' file that you provide.

That image should be mounted properly inside the guest. Go to it and launch the installer.

**Comment by ericrhenry5488** [comment]
It's been a long week. I soon figured out that I had loop-mounted the .iso on the Linux side to inspect its contents, and forgot to unmount it before trying to connect to it from the VM. Once I fixed that, I was able to do exactly what you described, and the install seems to have worked. It actually only offered the option to "Repair" the installation, which means that maybe it had somehow been properly installed, but I wasn't looking in the right place for it.

Anyway, I **think** things are sane now, Again, many thanks for your patience.
